### PR TITLE
fix(coverage): Add files to be included or excluded during code coverage generation

### PR DESCRIPTION
**Pull request by nyakamwe**
**What does this PR do?**
It fixes bugs related to code coverage status generation.

**Description of Task to be completed?**
- adding all js files found in src to be included
- exclude files from the test folder

**How should this be manually tested?**
After cloning this repo, add or remove some codes and run: npm run coverage
this will deliver a report to the coveralls and generate coverage status in percentage.

**What are the relevant pivotal tracker stories?**
#182209989

[coveralls #182209989]

### CODE REVIEW

**Comment by Himbaza123h** [comment]
Well done!

**Review by munezerobagira — DISMISSED** [review]

**Review by munezerobagira — COMMENTED** [review]
Checking  on coveralls the `src/sum.js` is only file which  has the coverage ,  you can increase the coverage to other files as well
https://coveralls.io/builds/49892434
image
 make sure that other files like src/routes/index.js are checked too

File: package.json [hunk 1]
```diff
@@ -4,8 +4,8 @@
   "main": "test.js",
   "scripts": {
     "build": "npx babel src --out-dir lib",
-    "test": "nyc --reporter=text --reporter=html mocha --recursive --require @babel/polyfill --require @babel/register src/test",
-    "coverage": "nyc  report --reporter=text-lcov --reporter=lcov",
+    "test": "mocha --recursive --require @babel/polyfill --require @babel/register ./src/test/*.js --timeout 30000 --exit",
+    "coverage": "nyc --reporter=text --reporter=lcov npm run test",
```
Comment (munezerobagira) [inline package.json]: you can add the --all after nyc to collect  coverage for the entire project

File: .nycrc.json [hunk 2]
```diff
@@ -3,5 +3,8 @@
     "statements": 80,
     "branches": 80,
     "functions": 80,
-    "lines": 80
+    "lines": 80,
+    "include": ["src/**/*.js"],
+    "exclude": ["src/**/*.test.js", "src/database/migrations/*.js"]
```
Comment (munezerobagira) [inline .nycrc.json]: I am not sure if we will write the tests for seeders too, if possible you can add it  to exclude too

Comment (K-Ds) [inline .nycrc.json]: You can also add **"src/database/seeders/*.js** to the list

File: package.json (hunk 1 above)
Comment (munezerobagira) [reply package.json]: Or you can add it in .nycrc.json

**Comment by nyakamwe** [comment]
The threshold is above 80% in local but when it comes to CI it decreases unexpectedly

**Comment by nyakamwe** [comment]
> Checking on coveralls the `src/sum.js` is only file which has the coverage , you can increase the coverage to other files as well https://coveralls.io/builds/49892434 image make sure that other files like src/routes/index.js are checked too
coveralls

**Review by munezerobagira — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (nyakamwe) [reply package.json]: yes, I have added it in .nycrc.json

File: .nycrc.json (hunk 2 above)
Comment (nyakamwe) [reply .nycrc.json]: seeders are also excluded.

**Review by istafrica-dennw — CHANGES_REQUESTED** [review]
@nyakamwe ,
Your PR has conflicts

**Review by habaumugisha1 — APPROVED** [review]

**Review by istafrica-dennw — APPROVED** [review]

**Pull request merged**